<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <title>Lehmann Lab</title>
  <link rel="icon" type="image/png" sizes="32x32" href="assets/media/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="assets/media/favicon-16x16.png">
  <meta name="description" content="Computational Biology Lab of the Cancer Center Cologne Essen at Uniklinik RWTH Aachen">

  <!-- The important stuff for appearance on social media -->
    <meta property="og:type" content="website" />
    <meta property="og:title" content="Lehmann Lab">
    <meta property="og:description" content="Computational Biology Lab of CCCE Uniklinik Köln and Uniklinik RWTH Aachen">
    <meta property="og:image"
      content="https://raw.githubusercontent.com/joyceykao/lehmannlabsite/master/assets/media/LLoglogo.png">
    <meta name="twitter:image"
      content="https://raw.githubusercontent.com/joyceykao/lehmannlabsite/master/assets/media/LLoglogo.png">
    <meta property="og:url" content="https:lehmannlab.org">
    <meta name="twitter:card" content="summary_large_image">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet"
    integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
  <link rel="stylesheet" href="assets/css/style.css">
</head>

<body>

  <nav class="navbar fixed-top navbar-expand-md navbar-dark">
    <div class="container-fluid">
      <a class="navbar-brand" href="./index.html">
        <img src="./assets/media/favicon-32x32.png" width="30" height="30" class="d-inline-block align-top" alt="">
        Lehmann Lab
      </a>
      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav"
        aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarNav">
        <ul class="navbar-nav ">
          <!-- <li class="nav-item">
            <a class="nav-link" href="./index.html">Home</a>
          </li> -->
          <li class="nav-item">
            <a class="nav-link" href="./research.html">Research</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="./code.html">Code</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="./people.html">People</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="./opportunities.html">Opportunities</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="./contact.html">Contact</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="https://twitter.com/lehmannlab"><i class="fab fa-twitter"></i></a>
          </li>
        </ul>
      </div>
    </div>
  </nav>

  <div class="content mt-5 mb-5">
    <div class="content-inside">
      <div class="container">
        <h1>Code</h1>
        <!-- Intro -->
        <div class="card mt-3">
          <div class="card-body">
            <h5 class="card-title">Repositories for frameworks</h5>
            <div class="card-text">
              The following code repositories were developed in parallel with the accompanying research publications. Please refer to
              publications for more details or <a href="mailt:kjlehmann@ukaachen.de">send us a message</a>.
            </div>
          </div>
        </div>

        <!-- Scatter -->
        <div class="card mt-3">
          <div class="card-body">
            <h5 class="card-title">SCATTER: Fast SNV-based single cell variant caller</h5>
            <a class="btn mb-2" target="_blank" href="https://github.com/ratschlab/scatter"><i class="fab fa-github"></i>
              Repository</a>
            <div class="accordion accordion-flush" id="accordionFlushExample">
              <div class="accordion-item">
                <div class="accordion-header" id="flush-headingOne">
                  <button class="accordion-button collapsed" id="code-accordion" type="button" data-bs-toggle="collapse"
                    data-bs-target="#flush-collapseTwo" aria-expanded="false" aria-controls="flush-collapseTwo">
                    <div>
                      <p>SCATTER (Single Cell dATa clusTERer) is able to cluster cells and perform variant calling based on information obtained from single-cell DNA
                      sequencing. SCATTER takes as input BAM files containing the aligned data for each cell and provides as output a
                      clustering of the cells and, optionally, VCF files pinpointing the changes relative to a reference genome.</p>
                    </div>
                  </button>
                </div>
                <div id="flush-collapseTwo" class="accordion-collapse collapse" aria-labelledby="flush-headingTwo"
                  data-bs-parent="#accordionFlushExample">
                  <div class="accordion-body">
                    <div class="publication">
                      <div class="publication-title">Clustering cells based on single-cell DNA-sequencing data with ultra-low coverage</div>
                      <div class="publication-place">Publication Coming Soon</div>
                    </div>

                    <a class="btn" target="_blank" href="https://github.com/ratschlab/scatter"><i class="fab fa-github"></i>
                      Repository</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!-- SCIM -->
        <div class="card mt-3">
          <div class="card-body">
            <h5 class="card-title">SCIM: Universal Single-Cell Matching with Unpaired Feature Sets</h5>
            <a class="btn mb-2" target="_blank" href="https://github.com/ratschlab/scim"><i class="fab fa-github"></i>
              Repository</a>
            <div class="accordion accordion-flush" id="accordionFlushExample">
              <div class="accordion-item">
                <div class="accordion-header" id="flush-headingOne">
                  <button class="accordion-button collapsed" id="code-accordion" type="button" data-bs-toggle="collapse"
                    data-bs-target="#flush-collapseOne" aria-expanded="false" aria-controls="flush-collapseOne">
                    <div>
                      <p>Integrates datasets from multiple single cell 'omics technologies in two steps:</p>
                      <ul>
                        <li>Constructs a technology invariant latent space</li>
                        <li>Matches cells across technologies by bipartite matching of latent representations</li>
                      </ul>
                    </div>
                  </button>
                </div>
                <div id="flush-collapseOne" class="accordion-collapse collapse" aria-labelledby="flush-headingOne"
                  data-bs-parent="#accordionFlushExample">
                  <div class="accordion-body">
                    <div class="publication">
                      <div class="publication-authors">Stefan G Stark, Joanna Ficek, Francesco Locatello, Ximena Bonilla,
                        Stéphane Chevrier, Franziska Singer, Tumor Profiler
                        Consortium, Gunnar Rätsch, <strong>Kjong-Van Lehmann</strong></div>
                      <div class="publication-title">SCIM: universal single-cell matching with unpaired feature sets</div>
                      <div class="publication-place">Bioinformatics</div>
                    </div>

                    <p><strong>Abstract:</strong><br>
                      <strong>Motivation</strong><br>
                      Recent technological advances have led to an increase in the production and availability of single-cell
                      data. The
                      ability to integrate a set of multi-technology measurements would allow the identification of biologically
                      or
                      clinically
                      meaningful observations through the unification of the perspectives afforded by each technology. In most
                      cases,
                      however,
                      profiling technologies consume the used cells and thus pairwise correspondences between datasets are lost.
                      Due to the
                      sheer size single-cell datasets can acquire, scalable algorithms that are able to universally match
                      single-cell
                      measurements carried out in one cell to its corresponding sibling in another technology are needed. <br>
                      <strong>Results</strong><br>
                      We propose Single-Cell data Integration via Matching (SCIM), a scalable approach to recover such
                      correspondences in
                      two
                      or more technologies. SCIM assumes that cells share a common (low-dimensional) underlying structure and
                      that the
                      underlying cell distribution is approximately constant across technologies. It constructs a
                      technology-invariant
                      latent
                      space using an autoencoder framework with an adversarial objective. Multi-modal datasets are integrated by
                      pairing
                      cells
                      across technologies using a bipartite matching scheme that operates on the low-dimensional latent
                      representations. We
                      evaluate SCIM on a simulated cellular branching process and show that the cell-to-cell matches derived by
                      SCIM reflect
                      the same pseudotime on the simulated dataset. Moreover, we apply our method to two real-world scenarios, a
                      melanoma
                      tumor sample and a human bone marrow sample, where we pair cells from a scRNA dataset to their sibling
                      cells in a
                      CyTOF
                      dataset achieving 90% and 78% cell-matching accuracy for each one of the samples, respectively.
                    </p>
                    <a class="btn" target="_blank"
                      href="https://www.cell.com/cancer-cell/fulltext/S1535-6108(21)00048-9">Publication</a>
                    <a class="btn" target="_blank" href="https://github.com/ratschlab/scim"><i class="fab fa-github"></i>
                      Repository</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!-- SNBNMF -->
        <div class="card mt-3">
          <div class="card-body">
            <h5 class="card-title">SNBNMF: Supervised Negative Binomial NMF for Mutational Signature Discovery</h5>
            <a class="btn mb-2" target="_blank" href="https://github.com/ratschlab/SNBNMF-mutsig-public"><i class="fab fa-github"></i>
              Repository</a>
            <div class="accordion accordion-flush" id="accordionFlushExample">
              <div class="accordion-item">
                <div class="accordion-header" id="flush-headingThree">
                  <button class="accordion-button collapsed" id="code-accordion" type="button" data-bs-toggle="collapse"
                    data-bs-target="#flush-collapseThree" aria-expanded="false" aria-controls="flush-collapseThree">
                    <p>Code to replicate our work to derive mutational signatures within a negative binomial framework. The matrix
                    factorization is coupled with a support vector machine to allow the use of labeled data.</p>
                  </button>
                </div>
                <div id="flush-collapseThree" class="accordion-collapse collapse" aria-labelledby="flush-headingThree"
                  data-bs-parent="#accordionFlushExample">
                  <div class="accordion-body">
                    <div class="publication">
                      <div class="publication-authors">Xinrui Lyu, Jean Garret, Gunnar Rätsch, <strong>Kjong-Van Lehmann</strong></div>
                      <div class="publication-title">Mutational signature learning with supervised negative binomial non-negative matrix
                        factorization</div>
                      <div class="publication-place">Bioinformatics</div>
                      <strong>Abstract:</strong>
                      <p><strong>Motivation</strong><br>
                        Understanding the underlying mutational processes of cancer patients has been a long-standing goal in the community
                        and
                        promises to provide new insights that could improve cancer diagnoses and treatments. Mutational signatures are
                        summaries
                        of the mutational processes, and improving the derivation of mutational signatures can yield new discoveries
                        previously
                        obscured by technical and biological confounders. Results from existing mutational signature extraction methods depend
                        on the size of available patient cohort and solely focus on the analysis of mutation count data without considering
                        the
                        exploitation of metadata. <br>
                        <strong>Results</strong><br>
                        Here we present a supervised method that utilizes cancer type as metadata to extract more distinctive signatures. More
                        specifically, we use a negative binomial non-negative matrix factorization and add a support vector machine loss. We
                        show that mutational signatures extracted by our proposed method have a lower reconstruction error and are designed to
                        be more predictive of cancer type than those generated by unsupervised methods. This design reduces the need for
                        elaborate post-processing strategies in order to recover most of the known signatures unlike the existing unsupervised
                        signature extraction methods. Signatures extracted by a supervised model used in conjunction with cancer-type labels
                        are
                        also more robust, especially when using small and potentially cancer-type limited patient cohorts. Finally, we adapted
                        our model such that molecular features can be utilized to derive an according mutational signature. We used APOBEC
                        expression and MUTYH mutation status to demonstrate the possibilities that arise from this ability. We conclude that
                        our
                        method, which exploits available metadata, improves the quality of mutational signatures as well as helps derive more
                        interpretable representations.</p>

                    </div>
                    <a class="btn" target="_blank" href="https://academic.oup.com/bioinformatics/article/36/Supplement_1/i154/5870496">Publication</a>
                    <a class="btn" target="_blank" href="https://github.com/ratschlab/SNBNMF-mutsig-public"><i class="fab fa-github"></i>
                      Repository</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!-- ICGC -->
        <div class="card mt-3">
          <div class="card-body">
            <h5 class="card-title">ICGC QTL Analysis</h5>
            <a class="btn mb-2" target="_blank" href="https://github.com/ICGC-TCGA-PanCancer/pcawg3_germline_qtl"><i
                class="fab fa-github"></i>
              Repository</a>
            <div class="accordion accordion-flush" id="accordionFlushExample">
              <div class="accordion-item">
                <div class="accordion-header" id="flush-headingThree">
                  <button class="accordion-button collapsed" id="code-accordion" type="button" data-bs-toggle="collapse"
                    data-bs-target="#flush-collapseFour" aria-expanded="false" aria-controls="flush-collapseFour">
                    <div>
                      <p>
                        Code for reproducing the QTL Analysis of PCAWG-ICGC work
                      </p>
                    </div>
                  </button>
                </div>
                <div id="flush-collapseFour" class="accordion-collapse collapse" aria-labelledby="flush-headingFour"
                  data-bs-parent="#accordionFlushExample">
                  <div class="accordion-body">
                    <div class="publication">
                      <div class="publication-authors">XPCAWG Transcriptome Core Group, Claudia Calabrese, Natalie R. Davidson, Deniz
                        Demircioğlu, Nuno A. Fonseca, Yao He,
                        André Kahles, <strong>Kjong-Van Lehmann</strong>, Fenglin Liu, Yuichi Shiraishi, Cameron M. Soulette, Lara Urban,
                        Liliana Greger, Siliang
                        Li, Dongbing Liu, Marc D. Perry, Qian Xiang, Fan Zhang, Junjun Zhang, Peter Bailey, Serap Erkek, Katherine A. Hoadley,
                        Yong Hou, Matthew R. Huska, Helena Kilpinen, Jan O. Korbel, Maximillian G. Marin, Julia Markowski, Tannistha Nandi,
                        Qiang Pan-Hammarström, Chandra Sekhar Pedamallu, Reiner Siebert, Stefan G. Stark, Hong Su, Patrick Tan, Sebastian M.
                        Waszak, Christina Yung, Shida Zhu, Philip Awadalla, Chad J. Creighton, Matthew Meyerson, B. F. Francis Ouellette, Kui
                        Wu, Huanming Yang, PCAWG Transcriptome Working Group, Alvis Brazma, Angela N. Brooks, Jonathan Göke, Gunnar Rätsch,
                        Roland F. Schwarz, Oliver Stegle, Zemin Zhang & PCAWG Consortium</div>
                      <div class="publication-title">Genomic basis for RNA alterations in cancer</div>
                      <div class="publication-place">Nature</div>
                      <strong>Abstract:</strong>
                      <p>Transcript alterations often result from somatic changes in cancer genomes1. Various forms of RNA alterations have
                        been described in cancer, including overexpression2, altered splicing3 and gene fusions4; however, it is difficult to
                        attribute these to underlying genomic changes owing to heterogeneity among patients and tumour types, and the
                        relatively small cohorts of patients for whom samples have been analysed by both transcriptome and whole-genome
                        sequencing. Here we present, to our knowledge, the most comprehensive catalogue of cancer-associated gene alterations
                        to date, obtained by characterizing tumour transcriptomes from 1,188 donors of the Pan-Cancer Analysis of Whole
                        Genomes (PCAWG) Consortium of the International Cancer Genome Consortium (ICGC) and The Cancer Genome Atlas (TCGA)5.
                        Using matched whole-genome sequencing data, we associated several categories of RNA alterations with germline and
                        somatic DNA alterations, and identified probable genetic mechanisms. Somatic copy-number alterations were the major
                        drivers of variations in total gene and allele-specific expression. We identified 649 associations of somatic
                        single-nucleotide variants with gene expression in cis, of which 68.4% involved associations with flanking non-coding
                        regions of the gene. We found 1,900 splicing alterations associated with somatic mutations, including the formation of
                        exons within introns in proximity to Alu elements. In addition, 82% of gene fusions were associated with structural
                        variants, including 75 of a new class, termed ‘bridged’ fusions, in which a third genomic location bridges two genes.
                        We observed transcriptomic alteration signatures that differ between cancer types and have associations with
                        variations in DNA mutational signatures. This compendium of RNA alterations in the genomic context provides a rich</p>

                    </div>
                    <a class="btn" target="_blank"
                      href="https://www.nature.com/articles/s41586-020-1970-0">Publication</a>
                    <a class="btn" target="_blank" href="https://github.com/ICGC-TCGA-PanCancer/pcawg3_germline_qtl"><i
                        class="fab fa-github"></i>
                      Repository</a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>

  <footer class="footer p-1">
    <div class="flex-row d-flex justify-content-between">
      <div class="container text-center">
        <a href="https://ccce.nrw/" target="_blank" rel="noopener noreferrer">
          <h6>CCCE</h6>
        </a>
      </div>
      <div class="container text-center">
        <p style="font-size: 10px;">©2021 Copyright Lehmann Lab</p>
      </div>
      <div class="container align-items-center">
        <a href="https://www.ukaachen.de/" target="_blank" rel="noopener noreferrer"><img class="mt-1"
            src="assets/media/logo-uniklinik-rwth-aachen.svg" style="max-height: 30px; width: auto;"></a>
      </div>
    </div>
  </footer>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"
    integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p"
    crossorigin="anonymous"></script>
</body>

</html>
